### PR TITLE
Python in CI/CD, add lintage and tests

### LINKED ISSUE #247: Protoc-generated Python code should pass pylint/black checks
This is to ensure consistent coding guidelines between https://github.com/in-toto/in-toto and the Python bindings provided in this repo.

### PR DESCRIPTION
**Pull request by woodruffw**
This adds a `python.yml` workflow, which contains both a test matrix and a linting job.

It also adds a convenience `Makefile` for performing common developer tasks on the Python binding.

Closes #247.

### CODE REVIEW

**Comment by woodruffw** [comment]
CC @adityasaky @joshuagl

**Review by joshuagl — COMMENTED** [review]
Thanks for working on this so promptly!

File: .github/workflows/python.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      - main
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  test:
+    strategy:
+      matrix:
+        python:
+          - "3.7"
```
[the remark this answers is not in the document]
Comment (woodruffw) [reply .github/workflows/python.yml:15]: Yep, 3.7 is EOL. The metadata currently supports it so I figured I'd add it here, and can remove it after a subsequent release. 

Alternatively, I can remote it outright if there's a preference for that 🙂

[the remark this answers is not in the document]
Comment (woodruffw) [reply .github/workflows/python.yml:15]: (None of the others are EOL yet, I believe.)

**Review by marcelamelara — APPROVED** [review]
Thanks for theses updates @woodruffw ! 

I think this also addresses/supersedes #247. So I'll close that once this is merged.

**Comment by marcelamelara** [comment]
@woodruffw I assume this branch may need to be rebased wrt main? Can you pls resolve the conflicts?

**Comment by woodruffw** [comment]
> @woodruffw I assume this branch may need to be rebased wrt main? Can you pls resolve the conflicts?

Done!

File: .github/workflows/python.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (joshuagl) [reply .github/workflows/python.yml:15]: My weak preference would be to remove it, but as the metadata already claims support for 3.7 that's out of scope for this PR. Thanks!

**Review by joshuagl — APPROVED** [review]
This is great, thank you!

**Pull request merged**